blocks/math/include/gnuradio-4.0/math/ExpressionBlocks.hpp

```diff
@@ -0,0 +1,352 @@
+        throw std::out_of_range(fmt::format("invalid vector boundaries [{}, {}]", base_ptr, end_ptr));
+    }
+
+    return {static_cast<std::size_t>(end - base) / elementSize, (access - base) / static_cast<ssize_t>(elementSize)};
+}
+
+struct vector_access_rtc : public exprtk::vector_access_runtime_check {
+    std::unordered_map<void*, std::string> vector_map;
+
+    bool handle_runtime_violation(violation_context& context) override {
+        auto               itr         = vector_map.find(static_cast<void*>(context.base_ptr));
+        const std::string& vector_name = (itr != vector_map.end()) ? itr->second : "Unknown";
+
+        const auto typeSize      = static_cast<std::size_t>(context.type_size);
+        auto [vecSize, vecIndex] = computeVectorInfo(context.base_ptr, context.end_ptr, typeSize, context.access_ptr);
+        throw gr::exception(fmt::format("vector access '{name}[{index}]' outside of [0, {size}[ (typesize: {typesize})", //
+            fmt::arg("name", vector_name), fmt::arg("size", vecSize), fmt::arg("index", vecIndex), fmt::arg("typesize", typeSize)));
+        return false; // should never reach here
+    }
+};
+
+} // namespace detail
+
+template<typename T>
+requires std::floating_point<T>
+struct ExpressionSISO : Block<ExpressionSISO<T>> {
+    using Description = Doc<R""(@brief Single-Input-Single-Output (SISO) expression evaluator.
+
+This block uses ExprTK to compute a user-defined expression for each input sample.
+The input sample is referenced by the variable `x`, and the output is produced as the evaluated expression.
+
+Examples:
+- `y := a * x + b`           // (simple linear scaling)
+- `a * x + b`                // (as above, the 'y:=' is optional)
+- `y := sin(pi * x)`         // (non-linear transformation)
+- `y := y + 0.1*x`           // (recursive IIR-like update using `y` as state)
+- `y := clamp(-1.0, x, 1.0)` // (clamping the input range)
+
+For full syntax, conditionals, loops, and advanced features:
+@see https://www.partow.net/programming/exprtk/index.html
+@see https://github.com/ArashPartow/exprtk
+)"">;
+    template<typename U, fixed_string description = "", typename... Arguments>
+    using A = Annotated<U, description, Arguments...>;
+
+    PortIn<T>  in;
+    PortOut<T> out;
+
+    A<std::string, "expr string", Doc<"for syntax see: https://github.com/ArashPartow/exprtk">> expr_string = "clamp(-1.0, sin(2 * pi * x) + cos(x / 2 * pi), +1.0)";
+    A<T, "a", Doc<"free parameter 'a' for use in expressions">, Visible>                        param_a     = T(1.0);
+    A<T, "b", Doc<"free parameter 'b' for use in expressions">, Visible>                        param_b     = T(0.0);
+    A<T, "c", Doc<"free parameter 'c' for use in expressions">, Visible>                        param_c     = T(0.0);
+
+    GR_MAKE_REFLECTABLE(ExpressionSISO, in, out, expr_string, param_a, param_b, param_c);
+
+    exprtk::symbol_table<T> _symbol_table;
+    exprtk::expression<T>   _expression;
```

Comment:
```suggestion
    exprtk::symbol_table<T> _symbol_table{};
    exprtk::expression<T>   _expression{};
```
allows to use the parent's constructors.